neo/SmartContract/ContractManifest.cs

```diff
@@ -0,0 +1,220 @@
+                {
+                    Hash = hash,
+                    EntryPoint = new ContractMethodDescription()
+                    {
+                        Name = "Main",
+                        Parameters = new ContractParameterDefinition[]
+                        {
+                            new ContractParameterDefinition()
+                            {
+                                 Name = "operation",
+                                 Type = ContractParameterType.String
+                            },
+                            new ContractParameterDefinition()
+                            {
+                                 Name = "args",
+                                 Type = ContractParameterType.Array
+                            }
+                        },
+                        ReturnType = ContractParameterType.Array
+                    },
+                    Events = new ContractActionDescription[0],
+                    Methods = new ContractMethodDescription[0]
+                },
+                Features = ContractPropertyState.NoProperty,
+                Groups = null,
+                SafeMethods = WildCardContainer<string>.CreateWildcard(),
+                Trusts = WildCardContainer<UInt160>.CreateWildcard()
+            };
+        }
+
+        /// <summary>
+        /// Return true if is allowed
+        /// </summary>
+        /// <param name="manifest">Manifest</param>
+        /// <param name="method">Method</param>
+        /// <returns>Return true or false</returns>
+        public bool CanCall(ContractManifest manifest, string method)
+        {
+            if (Groups != null && manifest.Groups != null && Groups.Any(a => manifest.Groups.Any(b => a.PubKey.Equals(b.PubKey))))
+            {
+                // Same group
+                return true;
+            }
+
+            if (manifest.Trusts != null && !manifest.Trusts.IsWildcard && !manifest.Trusts.Contains(Hash))
+            {
+                // null == * wildcard
+                // You don't have rights in the contract
+                return false;
+            }
+
+            return Permissions == null || Permissions.IsWildcard || Permissions.Any(u => u.IsAllowed(manifest.Hash, method));
+        }
+
+        /// <summary>
+        /// Parse ContractManifest from json
+        /// </summary>
+        /// <param name="json">Json</param>
+        /// <returns>Return Contract manifest</returns>
+        public static ContractManifest Parse(string json)
+        {
+            return JsonConvert.DeserializeObject<ContractManifest>(json, _jsonSettings);
+        }
+
+        /// <summary>
+        /// To json
+        /// </summary>
+        /// <param name="indented">Indented</param>
+        /// <returns>Return json string</returns>
+        public string ToJson(bool indented = false)
+        {
+            return JsonConvert.SerializeObject(this, indented ? Formatting.Indented : Formatting.None, _jsonSettings);
```

Comment:
Should we keep `ToJson()` implementation consistent with other code in NEO?

Example:

https://github.com/neo-project/neo/blob/1202bec53f05c4080ef8fb4bb95efbf1cb0a251b/neo/Network/P2P/Payloads/Transaction.cs#L175-L190